src/debug/HyprCtl.cpp

```diff
@@ -580,10 +580,20 @@ std::string devicesRequest(eHyprCtlOutputFormat format, std::string request) {
         "variant": "{}",
         "options": "{}",
         "active_keymap": "{}",
+        "capsLock": {},
+        "numLock": {},
         "main": {}
     }},)#",
                 (uintptr_t)k.get(), escapeJSONStrings(k->hlName), escapeJSONStrings(k->currentRules.rules), escapeJSONStrings(k->currentRules.model),
                 escapeJSONStrings(k->currentRules.layout), escapeJSONStrings(k->currentRules.variant), escapeJSONStrings(k->currentRules.options), escapeJSONStrings(KM),
+                (((xkb_keymap_mod_get_index(k->xkbKeymap, XKB_MOD_NAME_CAPS)) != XKB_MOD_INVALID &&
```

Comment:
can this please be made into a small lambda that takes a SP to the keyboard and the mod name, this is unreadable

put the lambda in the beginning of the func and use it for both json and normal

```cpp
auto getModState = [](SP<IKeyboard> keyboard, uint32_t xkbModName) -> std::string {
   /// ... bla bla bla
   return isEnabled ? "true" : "false";
};
```

then just `getModState(k, XKB_MOD_NAME_CAPS)` for example